### DISCUSSION TITLE
Configure access request for a site

### DISCUSSION THREAD

**Discussion by plamber** [pnp/pnpcore#1016] [Q&A] [answered]
Hello,
I am trying to change the access request options for a SharePoint site.

- [ ] I saw that the property of IWeb of UseAccessRequestDefault is read only. 
- [ ] I also tried to change the AccessRequestSiteDescription property of IWeb and apply a decription for the access request in question. Unfortunately, I am not able to update it at all. I can read the values without issue though. 

I am wondering if there are other solutions to change the settings using the SDK. In case this is not implemented, I would suggest putting the AccessRequestSiteDescription as a get only property as it was done with UseAccessRequestDefault.

Last question since I haven't found it in the docs. Is there a way to see what REST and/or Graph calls are performed when calling UpdateBatch or any operation? It would be helpful to see if the SDK is performing the calls I expect against the service.

Thank you, a lot, for your feedback,
Patrick

**Comment by plamber** [comment, marked as answer]
In case if it helps, both properties are controlled over REST with these endpoints:
_api/web/SetUseAccessRequestDefaultAndUpdate
_api/web/SetAccessRequestSiteDescriptionAndUpdate

**Reply by jansenbe** [reply]
@plamber : this is something that needs to be added, can you create an issue to track this need or submit a PR for it. https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core/Model/SharePoint/Core/Internal/Web.cs#L1964-L2002 shows a sample you can base yourself on.

Regarding the call logging, see https://pnp.github.io/pnpcore/using-the-sdk/basics-settings.html#settings-overview. If you bump to `Debug` you should the actual calls being outputted. Alternative approach is using Fiddler
